### PR TITLE
`jaeger::TUDPTransport::write` crash when `getaddrinfo` returns error

### LINKED ISSUE #1114: jaeger::TUDPTransport::write crash when wrong endpoint used
**Describe your environment** 

opentelemetry-cpp ci build image

**Steps to reproduce**
Describe exactly how to reproduce the error. Include a code sample if applicable.

run `jaeger` example with wrong endpoint e.g.:
`bazel-bin/examples/jaeger/example_jaeger host.docker.internal:16386`

**What is the expected behavior?**

no crash, ideally an error message or an exception

What did you expect to see?

no crash

**What is the actual behavior?**

no crash

What did you see instead?

crash

**Additional context**

### PR DESCRIPTION
**Pull request by esigo**
Fixes #1114 (issue)

## Changes

when `getaddrinfo` used in `jaeger::TUDPTransport::open` returns error, we still tend to dereference `server_addr_info_` in member methods while it is `null`.
The PR adds the null check to avoid `SIGSEGV`.
For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: exporters/jaeger/src/TUDPTransport.cc [hunk 1]
```diff
@@ -75,6 +75,10 @@ void TUDPTransport::close()
 
 uint32_t TUDPTransport::read(uint8_t *buf, uint32_t len)
 {
+  if (!server_addr_info_)
```
Comment (ThomsonTan) [inline exporters/jaeger/src/TUDPTransport.cc:78]: Thanks for the catch. Could we also add internal log here?

Comment (esigo) [reply exporters/jaeger/src/TUDPTransport.cc:78]: I was afraid that the log here would be too much if the the function is called often.
I add the log in a sec.

Comment (ThomsonTan) [reply exporters/jaeger/src/TUDPTransport.cc:78]: Ok, perhaps add it right after the call to `getaddrinfo`.

Comment (esigo) [reply exporters/jaeger/src/TUDPTransport.cc:78]: There is already a log there:

`[Error] File: exporters/jaeger/src/TUDPTransport.cc:57Jaeger Exporter: getaddrinfo failed with error: -2`

Shall I change the message?

**Review by lalitb — APPROVED** [review]

File: exporters/jaeger/src/TUDPTransport.cc (hunk 1 above)
Comment (ThomsonTan) [reply exporters/jaeger/src/TUDPTransport.cc:78]: This current message is fine.

**Review by ThomsonTan — APPROVED** [review]

**Pull request merged**